src/Form/Form.js

```diff
@@ -221,16 +221,25 @@ class Form {
         const selector = this.matching.cssSelector('SUBMIT_BUTTON_SELECTOR')
         const allButtons = /** @type {HTMLElement[]} */([...this.form.querySelectorAll(selector)])
 
-        const likelySubmitButton = allButtons.find(isLikelyASubmitButton)
-        if (likelySubmitButton) return [likelySubmitButton]
-
-        return allButtons.filter((button) => {
-            const content = button.textContent || ''
-            const ariaLabel = button.getAttribute('aria-label') || ''
-            const title = button.title || ''
-            // trying to exclude the little buttons to show and hide passwords
-            return !/password|show|toggle|reveal|hide/i.test(content + ariaLabel + title)
-        })
+        return allButtons
+            .filter(isLikelyASubmitButton)
+            .filter((button) => {
+                const content = button.textContent || ''
+                const ariaLabel = button.getAttribute('aria-label') || ''
+                const title = button.title || ''
```

Comment:
Should we add `button.value` here, too? Does that change the test results?

If we do add it, does it make sense to find a more elegant way to concatenate the strings rather than the `+ + +` below?

Reply:
Even better, what if we add these checks right into `isLikelyASubmitButton`? It could also have positive effects on the accuracy of the login/signup detection. Do you mind giving it a go and seeing if it improves things?

If there are no negative effects on the test cases I would do it anyway, to centralise the logic and spare a loop 😉.

Reply:
I've moved this into isLikelyASubmitButton so that value is used in both places; it didn't fix anything though.

> If we do add it, does it make sense to find a more elegant way to concatenate the strings rather than the + + + below?

`[content, ariaLabel, title].join("")`? It doesn't seem any nicer to me though.

Reply:
Yeah, I meant maybe moving it to a `const` above rather than inline. I would also add whitespace when concatenating. It may not be an issue now, but if we start using word boundaries in the regex it will cause unexpected failures.

Reply:
Yeah, good point. Done in fab18190a6afc510def8690712604007b114069a.